app/javascript/packages/document-capture/components/documents-step.jsx

```diff
@@ -47,16 +48,26 @@ function DocumentsStep({
    *
    * @type {DocumentSide[]}
    */
-  const documentSides = selfieCaptureEnabled ? ['front', 'back', 'selfie'] : ['front', 'back'];
+  const documentSides = ['front', 'back'];
```

Comment:
I see why you went this direction. I agree that we should be splitting things out into components. Instead of doing it this way, I'd recommend you return a component structure like this from this file. Some of my reasoning:
- With the approach I'm suggesting all the headers are in the same file.
- With this approach we avoid the need for another component that mostly passes props through.
- I think naming the various instances of <DocumentSideAcuantCapture> is a good idea.

```jsx
return (
    <>
      {flowpath === ...}
      <PageHeading>...
      <DocumentCaptureSubheader> // returns the h2 starting with "1."
      <TipList>...
      <DocumentFront> // a new component, wraps DocumentSideAcuantCapture
      <DocumentBack> // a new component, wraps DocumentSideAcuantCapture
      <SelfieCaptureSubheader> // returns the h2 starting with "2."
      <Selfie> // a new component, wraps DocumentSideAcuantCapture
      {isLastStep ...}
      {notReadySectionEnabled ...}
      {exisQuestionSectionEnabled ...}
      <Cancel>
    </>
)
```

And for the `<DocumentFront>`, `<DocumentBack>`, and `<Selfie>` components, something like this in this same file (probably don't need a new file for each of these components):
```jsx
const DocumentFront = ({registerField, onChange, onError}: {...new type...}) = {
  const side: DocumentSide = 'front'
  return (
    <DocumentSideAcuantCapture
          key={side}
          side={side}
          registerField={registerField}
          value={value[side]}
          onChange={onChange}
          errors={errors}
          onError={onError}
    />
  )
}
```

Reply:
@charleyf , refactored with HOC for document sides. Also I kept selfie part as a separate component, since it contains multiple parts, feels easier to test as a whole piece.

Reply:
> it feels easier to test as a whole piece

I can see that. I do still think that everything in `document-capture-selfie-capture.tsx` should move into `documents-step.jsx`. I think that if you move everything up into `documents-step.jsx` your tests will continue to work (with minor setup changes) in `documents-step-spec.jsx`?

A few pieces of evidence:
- The two `h2`s ( 1. ... and 2. ...) should appear in the same file since they're at the same level on the same page.
- The props you're sending to `DocumentCaptureSelfieCapture` are identical to the props received by `DocumentsStep`.
- The two tip lists have identical props: `SelfieTipList` and `DocumentTipList`
- The props for `SelfieSection` (in the existing section) are identical to the props for `Selfie` (in the new file)

Another way I'm thinking about this: I can't describe in words what the purpose of `document-capture-selfie-capture` is, or how it's different from what `documents-step` accomplishes.

Reply:
Revert how front, back sides rendered, otherwise it cause some document active element issues during testing.

Reply:
@charleyf agree most of the bullet points, and seems backtracked from previously <DocumentFront> components etc.

Also, in general it may be callsed `document-capture-selfi-section`? Anyway I can pull into document-steps, separate it because later we will have it with `review-issue-step` page.

Reply:
@charleyf , pulled all stuff in document step.

Reply:
Nice! Having all the `TipList`, `<h2>` and `<DocumentSideAcuantCapture>` in the same file makes it much clearer to me.

> and seems backtracked from previously components etc

I'd be interested to hear more about this. The pseudocode for my [suggested component structure](https://github.com/18F/identity-idp/pull/9746#discussion_r1427382838) with an example component is earlier in this thread. I'm happy to help you implement that, but I think the way the code is currently is close enough.

Reply:
@charleyf , yes basically following is the same with the HOC withProps, where the element focus seems not behaves the same with rendering in places, I too was baffled by it since the change has nothing to do with functionality, but that feels quite elusive and can consume quite some time.
```
const DocumentFront = ({registerField, onChange, onError}: {...new type...}) = {
  const side: DocumentSide = 'front'
  return (
    <DocumentSideAcuantCapture
          key={side}
          side={side}
          registerField={registerField}
          value={value[side]}
          onChange={onChange}
          errors={errors}
          onError={onError}
    />
  )
}
```

Reply:
✅  just approved, since I agree it doesn't necessarily make much sense to keep polishing this.

Agreed, I'm also quite surprised that extracting those components would change anything. Can you tell me what you mean about the element focus changing? Do you mean the outline or some other aspect? A screenshot would help too.

Reply:
@charleyf in the document-capture-spec test, checks for document.activeElement